### PR TITLE
Redeem Event Watcher

### PR DESCRIPTION
**Pull request by memosys**
covers even watcher cron only!

### CODE REVIEW

File: package-lock.json
Comment (plor) [inline package-lock.json:1]: Can we keep this out of the root level?

File: package.json
Comment (plor) [inline package.json:1]: This feels not great to me.

File: web-portal/backend/src/utils/utils.service.ts [hunk 1]
```diff
@@ -124,6 +137,41 @@ export class UtilsService {
...
+    const logs = await viemClient.getLogs({
+      event,
+      fromBlock: blockNumber - BigInt(1000),
+      toBlock: blockNumber,
+    });
+
+    const parsedLogs: IParsedLog[] = logs.map((log: any) => ({
+      tenantId: web3.utils
+        .toAscii(log?.args?._identifier!)
+        .replaceAll(`\x00`, ''),
```
Comment (plor) [inline web-portal/backend/src/utils/utils.service.ts:155]: What is going on here?

File: web-portal/backend/src/utils/utils.service.ts [hunk 2]
```diff
@@ -124,6 +137,41 @@ export class UtilsService {
     }
     const data = await res.json();
     return data;
-  };
+  }
+
+  @Cron(CronExpression.EVERY_MINUTE)
+  async watchEvent() {
```
Comment (plor) [inline web-portal/backend/src/utils/utils.service.ts:143]: How does this know which chains and contracts to watch?

File: web-portal/backend/src/utils/viemClient.ts [hunk 3]
```diff
@@ -0,0 +1,7 @@
+import { createPublicClient, http } from 'viem';
+import { optimism } from 'viem/chains';
+
+export const viemClient = createPublicClient({
+  chain: optimism,
```
Comment (plor) [inline web-portal/backend/src/utils/viemClient.ts:5]: What is the best way to update this if we add new chains?

File: web-portal/backend/src/utils/utils.service.ts (hunk 2 above)
Comment (plor) [reply web-portal/backend/src/utils/utils.service.ts:143]: I see where chain is defined, still wondering about contracts though.

**Review by plor — CHANGES_REQUESTED** [review]
Concerned about some of the js bleed into root, and other comments are just curiosity mainly.

File: web-portal/backend/src/utils/viemClient.ts (hunk 3 above)
Comment (memosys) [reply web-portal/backend/src/utils/viemClient.ts:5]: Array of chains

File: web-portal/backend/src/utils/utils.service.ts (hunk 1 above)
Comment (memosys) [reply web-portal/backend/src/utils/utils.service.ts:155]: Just in case somebody does redeem from etherscan with padded zeros to right

File: package.json
Comment (memosys) [reply package.json:1]: Can remove

File: package-lock.json
Comment (memosys) [reply package-lock.json:1]: Yes

**Pull request merged**